client/hijack.go

```diff
@@ -175,15 +216,30 @@ func (cli *Client) setupHijackConn(req *http.Request, proto string) (net.Conn, e
 	defer clientconn.Close()
 
 	// Server hijacks the connection, error 'connection closed' expected
-	resp, err := clientconn.Do(req)
-	if err != httputil.ErrPersistEOF {
+	errCh := make(chan error, 1)
+	go func() {
+		resp, err := clientconn.Do(req)
```

Comment:
I'm not sure I understand this. Why doesn't `req.WithContext` already handle this?

Reply:
sadly, the `*httputil.ClientConn` methods do not respect the request context, per https://golang.org/src/net/http/httputil/persist.go?s=11092:11159#L415 -- there is a loud note on its factory method about how it should not be part of stdlib and a user should use methods in the `net/http` package and this library like won't be updated, it seems. I briefly investigated the possibility of using `net/http`, but there's no clear path using `net/http` methods to do what we want here.